### PR TITLE
Attempted fix of mop

### PR DESCRIPTION
**Pull request by rayshan**
Hi @marchant, can you try this branch? It doesn't seem to exit randomly for me on mac or linux now. The resulting build works on mac, but on linux it gives below errors:

```
GET http://192.168.99.100:4000/packages/montage@6527381/node_modules/bluebird/js/browser/bluebird.min.js 
browser.load @ montage.js:586
browser.bootstrap @ montage.js:699
exports.initMontage @ montage.js:54
(anonymous function) @ montage.js:891
(anonymous function) @ montage.js:39
(anonymous function) @ montage.js:41
```

And 2 similar errors, one for `http://192.168.99.100:4000/packages/montage@6527381/node_modules/mr/require.js`, another for `http://192.168.99.100:4000/packages/montage@6527381/node_modules/mr/browser.js`. Looks like rewriting of paths didn't work somewhere. I'll keep looking tomorrow.

### CODE REVIEW

**Comment by rayshan** [comment]
PR is updated with working mop, tested on mac and linux. Note the tests are failing due to tests using `q-io/mockFs`, but things actually do work. We need to reduce test dependencies on `q-io` as well.

File: package.json [hunk 1]
```diff
@@ -15,9 +15,9 @@
   },
   "dependencies": {
     "montage": ">=0.16.0",
-    "mr": "~0.16.0",
-    "collections": "~1.2",
-    "bluebird": "~3.0",
+    "mr": "~16.0.x",
+    "collections": "montagejs/collections#b1060f46a8ba69e200a34b02829ac93825926480",
```
Comment (marchant) [inline package.json]: Please set dependency to ~5.0.x

File: lib/bundle.js [hunk 2]
```diff
@@ -17,40 +16,40 @@ var bundleMontageHtml = require("./bundle/montage");
...
 // searches for oppotunities to bundle, then uses the appropriate bundler for
 // either Montage or stand-alone Mr.
 function bundleHtml(file, config) {
-    var package = file.package;
-    var montagePackage, montageLocation;
+    var package = file.package,
+        montagePackage, montageLocation,
+        requirePackage, requireLocation,
+        document,
+        jobs = [];
```
Comment (marchant) [inline lib/bundle.js]: Doesn't looks like jobs is being used?

**Comment by marchant** [comment]
Tested, merging in, couple of test issues to resolve

**Pull request merged**